dags/dynamic-challenge-dag.py

```diff
@@ -0,0 +1,141 @@
+import uuid
+from datetime import datetime
+
+import pandas as pd
+
+from airflow.decorators import dag, task
+from airflow.models import Param
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+from orca.services.nextflowtower import NextflowTowerHook
+from orca.services.nextflowtower.models import LaunchInfo
+from orca.services.synapse import SynapseHook
+
+UUID = uuid.uuid4()
+REGION_NAME = "us-east-1"
+BUCKET_NAME = "dynamic-challenge-project-tower-scratch"
+FILE_NAME = f"submissions_{UUID}.csv"
+KEY = "10days/dynamic_challenge"
+
+
+dag_params = {
+    "synapse_conn_id": Param("SYNAPSE_ORCA_SERVICE_ACCOUNT_CONN", type="string"),
+    "aws_conn_id": Param("AWS_TOWER_PROD_S3_CONN", type="string"),
+    "tower_conn_id": Param("DYNAMIC_CHALLENGE_PROJECT_TOWER_CONN", type="string"),
+    "tower_run_name": Param(f"dynamic-challenge-evaluation_{UUID}", type="string"),
+    "tower_compute_env_type": Param("spot", type="string"),
+    "view_id": Param("syn52658661", type="string"),
+    "testing_data": Param("syn53627077", type="string"),
+    "scoring_script": Param("dynamic_challenge_score.py", type="string"),
+    "validation_script": Param("dynamic_challenge_validate.py", type="string"),    
+    "email_with_score": Param("yes", type="string"),
+    "synapse_evaluation_id": Param("9615537", type="string"),
+}
+
+dag_config = {
+    # Run every 2 minutes
+    "schedule_interval": "*/2 * * * *",
+    "start_date": datetime(2024, 1, 1),
+    "catchup": False,
+    "default_args": {
+        "retries": 2,
+    },
+    "tags": ["nextflow_tower"],
+    "params": dag_params,
+}
+
+
+@dag(**dag_config)
+def dynamic_challenge_dag():
+    @task.branch()
+    def check_for_new_submissions(**context):
+        hook = SynapseHook(context["params"]["synapse_conn_id"])
+        # Unfortunately, you cannot use XCom in a branching task
+        # So we have to call `get_submissions_with_status` twice in the DAG
+        submissions = hook.ops.get_submissions_with_status(
```

Comment:
@jaymedina Something I noticed in https://github.com/Sage-Bionetworks-Workflows/py-orca/blob/main/src/orca/services/synapse/ops.py#L97 - The way this is set up:
`f"select * from {submission_view} where status = '{submission_status}'"` means that you are grabbing all columns of the table, but right after this you are only getting the ID column in the DF:

`submission_ids = query_results.asDataFrame()["id"].tolist()`

The select statement should change to:

`f"select id from {submission_view} where status = '{submission_status}'"

The reason is that only the ID would then be transferred over HTTP, instead of all columns - It's minor, but best practice to only transfer the minimum amount of data required. Do you mind creating a tech debt item to tackle this change?

Reply:
Made the ticket, thanks @BryanFauble : https://sagebionetworks.jira.com/browse/IBCDPE-848